allennlp/modules/transformer/transformer_embeddings.py

```diff
@@ -7,132 +7,153 @@
+        The probability of an element to be zeroed.
+    """
+
+    def __init__(self, feature_size: int, embedding_size: int, dropout: float = 0.0):
+        image_embeddings = torch.nn.Linear(feature_size, embedding_size)
+        location_embeddings = torch.nn.Linear(4, embedding_size)
+        embeddings = torch.nn.ModuleDict(
+            {"image_embeddings": image_embeddings, "location_embeddings": location_embeddings}
+        )
+        super().__init__(embeddings, embedding_size, dropout)
+
+
+class TransformerEmbeddings(Embeddings):
     """
     Construct the embeddings from word, position and token_type embeddings.
     Details in the paper:
     [BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding, Devlin et al, 2019]
     (https://api.semanticscholar.org/CorpusID:52967399)
+
+    # Parameters
+
+    vocab_size : `int`
+        The size of the input vocab.
+    embedding_size : `int`
+        The `embedding_dim` of all the embedding layers.
+    pad_token_id : `int` (default = `0`)
+        The token id of the `<pad>` token.
+    max_position_embeddings : `int` (default = `512`)
+        The maximum number of positions.
+    type_vocab_size : `int` (default = `2`)
+        The size of the input token_type vocab.
+    dropout : `int` (default = `0.1`)
+        The probability of an element to be zeroed.
+    output_size : `int`, optional (default = `None`)
+        Optionally apply a linear transform after the dropout, projecting to `output_size`.
     """
 
     _relevant_module = "embeddings"
-    _huggingface_mapping = {"LayerNorm": "embeddings.layer_norm", "dropout": "embeddings.dropout"}
+    _huggingface_mapping = {
+        "LayerNorm": "layer_norm",
+        "word_embeddings": "embeddings.word_embeddings",
+        "position_embeddings": "embeddings.position_embeddings",
+        "token_type_embeddings": "embeddings.token_type_embeddings",
+    }
 
     def __init__(
         self,
         vocab_size: int,
-        hidden_size: int,
+        embedding_size: int,
         pad_token_id: int = 0,
         max_position_embeddings: int = 512,
         type_vocab_size: int = 2,
         dropout: float = 0.1,
+        output_size: Optional[int] = None,
     ):
-        super().__init__()
-        word_embeddings = torch.nn.Embedding(vocab_size, hidden_size, padding_idx=pad_token_id)
-        position_embeddings = torch.nn.Embedding(max_position_embeddings, hidden_size)
-        token_type_embeddings = torch.nn.Embedding(type_vocab_size, hidden_size)
-
-        embeddings = [word_embeddings, position_embeddings, token_type_embeddings]
-
-        self.embeddings = Embeddings(embeddings, hidden_size, dropout)
-
-    def forward(
+        word_embeddings = torch.nn.Embedding(vocab_size, embedding_size, padding_idx=pad_token_id)
+        position_embeddings = torch.nn.Embedding(max_position_embeddings, embedding_size)
+        token_type_embeddings = torch.nn.Embedding(type_vocab_size, embedding_size)
```

Comment:
This works OK if `type_vocab_size` is 0? Not all transformers have types.

Reply:
It works now.